### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by taylorjohnson141**
### Code Highlights:
Adds error handling for common errors as well as makes the userID persist through storage.
 
### Where should the reviewer start?
 Calendar.tsx
 
### Have Tests Been Added?
- [x] No.
- [ ] Yes, but all tests are not passing.
- [ ] Yes, and all are passing.

### Any background context you want to provide?
Need to merge styling and tests from other dev branches.

### Issues:
* Addresses #70 
* Closes #

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: src/Components/FormStartPage/FormStartPage.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ type FormStartPageProps = {
 };
 let FormStartPage = (props: FormStartPageProps) => {
   useEffect(() => {
+    sessionStorage.setItem("userId",JSON.stringify(props.userId))
```
Comment (taylorjohnson141) [inline src/Components/FormStartPage/FormStartPage.tsx:11]: Setting User ID in session storage to allow for user to refresh

File: src/Components/CalendarComponent/CalendarComponent.tsx [hunk 2]
```diff
@@ -46,19 +45,27 @@ export const Get_User = gql`
...
+  if(props.userId ===0 || props.userId ===null && sessionStorage.getItem('userId') !== undefined ){
+    //@ts-ignore
+    id =JSON.parse(sessionStorage.getItem('userId'))
+  }  
+
   const { data, loading, error } = useQuery(Get_User, {
-    variables: { id: props.userId }
+    variables: { id: id }
   });
+  if(error && !id ){
```
Comment (taylorjohnson141) [inline src/Components/CalendarComponent/CalendarComponent.tsx:58]: If the user does not have an id and there has been an error

File: src/Components/Question/Question.tsx [hunk 3]
```diff
@@ -87,7 +87,7 @@ export const Question = (props: QuestionProps) => {
       >
         Skip
       </button>
-      <button
+     {date.length === 10 && (props.questionType === 'event' || endDate.length ===10) &&  <button
```
Comment (taylorjohnson141) [inline src/Components/Question/Question.tsx:90]: Forces user to inout both dates

File: src/Components/FormContainer/FormContainer.tsx [hunk 4]
```diff
@@ -29,6 +29,11 @@ export const getQuestionsQuery = gql`
 `;
 
 export const FormContainer = (props: FormContaineProps) => {
+  let id = props.userId
```
Comment (aemcdonald) [inline src/Components/FormContainer/FormContainer.tsx:32]: Awesome job error handling here incase the user refreshes the page, that makes for a much better UX

File: src/Components/FormStartPage/FormStartPage.tsx (hunk 1 above)
Comment (aemcdonald) [reply src/Components/FormStartPage/FormStartPage.tsx:11]: Awesome job error handling here incase the user refreshes the page, that makes for a much better UX

File: src/Components/Question/Question.test.tsx [hunk 5]
```diff
@@ -47,11 +47,11 @@ describe('Question', () => {
 
     const dateInput = getByTestId('date');
     const skipBtn = getByText('Skip');
-    const nextBtn = getByText('Next');
+    // const nextBtn = getByText('Next');
```
Comment (aemcdonald) [inline src/Components/Question/Question.test.tsx:50]: I'll make sure to fix these later!

File: src/Components/Question/Question.tsx (hunk 3 above)
Comment (aemcdonald) [reply src/Components/Question/Question.tsx:90]: Great error handling here as well!

**Review by aemcdonald — APPROVED** [review]
Awesome job with error handling here! Makes for a much better user experience. I'll make sure to fix the tests once this PR is merged.

**Review by krogowsk531 — APPROVED** [review]
Thanks for all the error handling!  So important!

**Pull request merged**